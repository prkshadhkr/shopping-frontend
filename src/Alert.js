import React from 'react';

const Alert = ({type, msgs}) => {
  return(
    <div className={`alert alert-${type}`} role="alert">
        {msgs.map(e => (
          <p className="mb-0 small" key={e}>
              {e}
          </p>))
        }
    </div>
  );
}

Alert.defaultProps = {
  type: "danger",
  msgs: []
}

export default Alert;