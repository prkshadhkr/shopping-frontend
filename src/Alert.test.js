import React from 'react';
import { render } from '@testing-library/react';
import Alert from './Alert';

it ('renders without crashing', function() {
  render(<Alert />);
});

it("matches snapshot for danger", function() {
  let messages = ['Nothing is working', 'Up hill battle'];
  const { asFragment } = render(<Alert type="danger" messages={messages} />);
  expect(asFragment()).toMatchSnapshot();
});

it ("matches snapshot for success", function() {
  let messages = ['Things look much better'];
  const { asFragment } = render(<Alert type="success" messages={messages} />);
  expect(asFragment()).toMatchSnapshot();
});