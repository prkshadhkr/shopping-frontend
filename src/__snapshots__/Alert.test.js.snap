// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for danger 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger"
    role="alert"
  />
</DocumentFragment>
`;

exports[`matches snapshot for success 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
    role="alert"
  />
</DocumentFragment>
`;
